src/tools/compiletest/src/runtest.rs

```diff
@@ -1816,12 +1826,35 @@ impl<'test> TestCx<'test> {
         }
     }
 
-    fn dump_output(&self, out: &str, err: &str) {
+    fn dump_output(&self, print_output: bool, proc_name: &str, out: &str, err: &str) {
         let revision = if let Some(r) = self.revision { format!("{}.", r) } else { String::new() };
 
         self.dump_output_file(out, &format!("{}out", revision));
         self.dump_output_file(err, &format!("{}err", revision));
-        self.maybe_dump_to_stdout(out, err);
+
+        if !print_output {
+            return;
+        }
+
+        let path = Path::new(proc_name);
+        let proc_name = if path.file_stem() == Some(OsStr::new("rmake")) {
```

Comment:
This can be `path.file_stem().is_some_and(|p| p == "rmake")`. But also, this will be different on Windows which will be `rmake.exe`, right?

Reply:
no. `file_stem` does not include the file extension: https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.file_stem

Reply:
Ah right, of course.